src/niamoto/core/plugins/transformers/ecological/custom_calculator.py

```
@@ -0,0 +1,1409 @@
+            # Add user variables
+            safe_dict.update(variables)
+
+            # Evaluate the formula
+            result = eval(formula, {"__builtins__": {}}, safe_dict)
+
+            # Convert result to a standard Python type if it's a NumPy type
+            if isinstance(result, np.ndarray):
+                if result.size == 1:
+                    result = result.item()  # Convert single-item array to scalar
+                else:
+                    result = result.tolist()  # Convert array to list
```

Ruff (0.8.2)

Use ternary operator `result = result.item() if result.size == 1 else result.tolist()` instead of `if`-`else`-block

(SIM108)